metric_tank/defcache.go

```diff
@@ -97,15 +99,23 @@ func (dc *DefCache) addToES(mdef *schema.MetricDefinition) {
 	} else {
 		metricsToEsOK.Inc(1)
 		dc.Lock()
-		dc.defs[mdef.Id] = mdef
+		dc.defsById[mdef.Id] = mdef
+		dc.defsByKey[mdef.Name] = mdef
```

Comment:
mdef.Name is not gobally unique.  As soon as you have to users try and write the same metric. ie 'litmus.google_com.new-york.ping.avg' you will have a collision.